plugins/module_utils/foreman_helper.py

```diff
@@ -444,8 +444,14 @@ def _delete_entity(self, resource, current_entity, params):
         payload = {'id': current_entity['id']}
         if params:
             payload.update(params)
-        self.resource_action(resource, 'destroy', payload)
-        return True, None
+
+        changed, entity = self.resource_action(resource, 'destroy', payload)
+
+        # this is a workaround for https://projects.theforeman.org/issues/26937
+        if entity and 'error' in entity and 'message' in entity['error']:
```

Comment:
Is this generic enough to have it in resource_action?

Reply:
this is at least true for location and organization, as both are a taxonomy, and the bug linked is affecting all taxonomies

Reply:
Sorry, i meant whether it can happen on other actions than delete, and whether we should employ it for create, update, you name it.

Reply:
ah, I wasn't sure about that. and the json *this* API endpoint returns on errors is slightly different then all the other endpoints, so let's keep it in delete for now.